CONTRIBUTING.md

```diff
@@ -89,6 +92,60 @@ Use the test scripts provided in the `ci/` directory to run tests for each compo
 ./ci/test_cub.sh -cxx g++ -std 14 -arch "70;75;80-virtual"
 ```
 
+### Using CMake Presets
+
+[CMake Presets](https://cmake.org/cmake/help/latest/manual/cmake-presets.7.html) are a set of configurations defined in a JSON file that specify project-wide build details for CMake. They provide a standardized and sharable way to configure, build, and test projects across different platforms and development environments. Presets are available from CMake version 3.19 and later.
+
+There are three kinds of Presets:
+
+- Configure Presets: specify options for the `cmake` command.
+
+- Builds Presets: specify options for the `cmake --build` command.
+
+- Test Presets: specify options for the `ctest` command.
+
+In CCCL we provide many presets to be used out of the box. You can find the complete list in the our corresponding [CMakePresets.json](./CMakePresets.json) file.
+
+#### Using CMake Presets via Command Line
+
+Once you have created your `build` directory and navigated into it, you can configure your project for a specific preset (e.g. `thrust-cpp11`) by executing the following command:
+
+```bash
+cmake --preset=thrust-cpp11 <path/to/cccl/root>
+```
+
+This command configures the project using the thrust-cpp11 preset. Please replace `<path/to/cccl/root>` with the actual path to the root directory of your CCCL project.
+
+Upon successful configuration, initiate the build process with:
+
+```bash
+cmake --build .
+```
+
+#### Using CMake Presets via VS Code GUI extension (Recommended when using DevContainers)
+
+The recommended way to use CMake Presets is via the VS Code extension [CMake Tools](https://marketplace.visualstudio.com/items?itemName=ms-vscode.cmake-tools). As soon as you install the extension you would be able to see the sidebar menu below.
+
+   ![cmaketools sidebar](/.devcontainer/img/cmaketools_sidebar.png)
+
+You can specify the desire CMake Preset by clicking the "Select Configure Preset" button under the "Configure" drop down option (see image below).
+
+   ![cmaketools presets](.devcontainer/img/cmaketools_presets.png)
+
+After that you can select the default build target from the "Build" drop down option. As soon as you click the button a drop down menu will appear with all the available targets that are included within the preset you selected. For example if you had selected the `all-dev` preset VS Code will display all the available targets we have in cccl.
```

Comment:
The screenshot looks like a tree widget to me, so I would use the terms node or item, instead of drop down menu.
```suggestion
After that you can select the default build target from the "Build" node. As soon as you expand it a list will appear with all the available targets that are included within the preset you selected. For example if you had selected the `all-dev` preset VS Code will display all the available targets we have in cccl.
```